rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,124 @@
+
+This build system became clunky and plastered, and consequently hard to understand, modify, improve and port to other systems besides Linux.
+
+Such state of things hinders development, specially outside the Linux and NixOS niches.
+
+In light of this, we propose a novel, from-scratch alternative build infrastructure.
+
+We expect to accomplish, among other goals,
+
+- better code structuring;
+- improved cross-platform support, especially in other programming environments, including but not limited to Unix-like operating systems;
+- shorter build times;
+- an overall improved user experience.
+
+# Detailed design
+[design]: #detailed-design
+
+A carefully crafted set of files written in Meson should be included in the Nix repository, in order to describe how to deploy the Nix repository, generating all the expected artifacts (command line tools, libraries, configuration files etc.)
+
+This novel build infrastructure should be able to provide at least all the features already present on the current quasi-autotools implementation, possibly with a different user interface.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Currently @p01arst0rm is writing an implementation from scratch.
+
+Some typical expected interactions are:
+
+- Configuring: `meson setup build_dir`
+- Building: `ninja -C build_dir` (or the backend-agnostic `meson -C build_dir compile`)
+- Installing: `ninja -C build_dir install` (or the backend-agnostic `meson -C build_dir install`)
+
+Usually, commandline options assume the format `-Dname=value`. This small change in user interface when compared to the typical `--name=value` is mostly harmless.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some possible drawbacks:
+
+- A new build system would require changes on the code
+  + On the other hand, such changes are likely to improve the code base.
+  
+- A new build system requires the developers become familiarized with it
+  - Specially when this build system uses its own description language.
+  + However, Meson is well documented, and its language is easy to grasp, specially for those familiarized with Python.
+
+- A new build system indirectly brings its own dependencies to the Nix project
+  - In particular, the reference implementation of Meson is written in Python.
+  - Further, this reference implementation generates script files meant to be consumed by Ninja, a tool written in C++ that acts like a Make replacement.
+  - This particular setting brings concerns about complexifying the bootstrap route.
+  + Given that Nix is currently written in C++, we can assume a C++ compiler as part of such a bootstrap route.
+  + There are full-featured alternative tools that replace Meson and Ninja. 
+    + Namely, Muon and Samurai are implementations of Meson and Ninja that require only a C compiler and a set of POSIX standard tools.
```

Comment:
consider adding:

"+ additionally, the reference meson implementation is capable of generating build files that target MSVC or XCode"